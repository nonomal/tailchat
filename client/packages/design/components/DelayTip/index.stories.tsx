import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { DelayTip } from '.';

export default {
  title: 'Tailchat/DelayTip',
  component: DelayTip,
  argTypes: {},
} as ComponentMeta<typeof DelayTip>;

const Template: ComponentStory<typeof DelayTip> = (args) => (
  <DelayTip {...args}>鼠标移动到这里等待1s</DelayTip>
);

export const Default = Template.bind({});
Default.args = {
  title: 'Hello World',
};
