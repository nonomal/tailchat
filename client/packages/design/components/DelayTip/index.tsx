import { Tooltip, TooltipProps } from 'antd';
import React from 'react';

/**
 * 延时提示
 */
export const DelayTip: React.FC<TooltipProps> = React.memo((props) => {
  return (
    <Tooltip mouseEnterDelay={1} {...props}>
      {props.children}
    </Tooltip>
  );
});
DelayTip.displayName = 'DelayTip';
