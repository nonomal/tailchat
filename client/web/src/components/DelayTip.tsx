export { DelayTip } from 'tailchat-design';
