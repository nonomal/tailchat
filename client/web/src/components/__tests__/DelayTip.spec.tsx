import { fireEvent, render, act } from '@testing-library/react';
import React from 'react';
import { sleep } from 'tailchat-shared';
import { DelayTip } from '../DelayTip';

describe('DelayTip', () => {
  test('render', async () => {
    const wrapper = render(
      <DelayTip title="any tip">
        <div data-testid="inner" />
      </DelayTip>
    );
    expect(wrapper.container).toMatchSnapshot('default');

    act(() => {
      fireEvent.mouseEnter(wrapper.getByTestId('inner'));
    });

    await sleep(500);
    expect(wrapper.container).toMatchSnapshot('not open');

    await sleep(1500);
    expect(wrapper.container).toMatchSnapshot('open');
  });
});
