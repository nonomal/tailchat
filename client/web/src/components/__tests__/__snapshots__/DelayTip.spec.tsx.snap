// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DelayTip render: default 1`] = `
<div>
  <div
    data-testid="inner"
  />
</div>
`;

exports[`DelayTip render: not open 1`] = `
<div>
  <div
    data-testid="inner"
  />
</div>
`;

exports[`DelayTip render: open 1`] = `
<div>
  <div
    class="ant-tooltip-open"
    data-testid="inner"
  />
</div>
`;
